plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableMetaQuery.php

```diff
@@ -0,0 +1,566 @@
+			$meta_compare_string_start .= "WHERE {$subquery_alias}.order_id = {$alias}.order_id ";
+			$meta_compare_string_end    = 'LIMIT 1';
+			$meta_compare_string_end   .= ')';
+		}
+
+		switch ( $clause['compare_key'] ) {
+			case '=':
+			case 'EXISTS':
+				$where = $wpdb->prepare( "$alias.meta_key = %s", trim( $clause['key'] ) ); // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+				break;
+			case 'LIKE':
+				$meta_compare_value = '%' . $wpdb->esc_like( trim( $clause['key'] ) ) . '%';
+				$where              = $wpdb->prepare( "$alias.meta_key LIKE %s", $meta_compare_value ); // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+				break;
+			case 'IN':
+				$meta_compare_string = "$alias.meta_key IN (" . substr( str_repeat( ',%s', count( $clause['key'] ) ), 1 ) . ')';
+				$where               = $wpdb->prepare( $meta_compare_string, $clause['key'] ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+				break;
+			case 'RLIKE':
+			case 'REGEXP':
+				$operator = $clause['compare_key'];
+				if ( isset( $clause['type_key'] ) && 'BINARY' === strtoupper( $clause['type_key'] ) ) {
+					$cast = 'BINARY';
+				} else {
+					$cast = '';
+				}
+				$where = $wpdb->prepare( "$alias.meta_key $operator $cast %s", trim( $clause['key'] ) ); // phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+				break;
+			case '!=':
+			case 'NOT EXISTS':
+				$meta_compare_string = $meta_compare_string_start . "AND $subquery_alias.meta_key = %s " . $meta_compare_string_end;
+				$where               = $wpdb->prepare( $meta_compare_string, $clause['key'] ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+				break;
+			case 'NOT LIKE':
+				$meta_compare_string = $meta_compare_string_start . "AND $subquery_alias.meta_key LIKE %s " . $meta_compare_string_end;
+
+				$meta_compare_value = '%' . $wpdb->esc_like( trim( $clause['key'] ) ) . '%';
+				$where              = $wpdb->prepare( $meta_compare_string, $meta_compare_value ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+				break;
+			case 'NOT IN':
+				$array_subclause     = '(' . substr( str_repeat( ',%s', count( $clause['key'] ) ), 1 ) . ') ';
+				$meta_compare_string = $meta_compare_string_start . "AND $subquery_alias.meta_key IN " . $array_subclause . $meta_compare_string_end;
+				$where               = $wpdb->prepare( $meta_compare_string, $clause['key'] ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+				break;
+			case 'NOT REGEXP':
+				$operator = $clause['compare_key'];
+				if ( isset( $clause['type_key'] ) && 'BINARY' === strtoupper( $clause['type_key'] ) ) {
+					$cast = 'BINARY';
+				} else {
+					$cast = '';
+				}
+
+				$meta_compare_string = $meta_compare_string_start . "AND $subquery_alias.meta_key REGEXP $cast %s " . $meta_compare_string_end;
+				$where               = $wpdb->prepare( $meta_compare_string, $clause['key'] ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+				break;
```

Comment:
perhaps add a `default` case in which `$where = '';`.

Reply:
✔️